gui/src/main/java/org/verapdf/gui/PDFValidationApplication.java

```diff
@@ -2,85 +2,87 @@
 
 import org.apache.log4j.Logger;
 import org.verapdf.gui.config.Config;
-import org.verapdf.gui.config.ConfigPropertiesSerializator;
+import org.verapdf.gui.tools.ConfigIO;
 import org.verapdf.gui.tools.GUIConstants;
 
 import javax.swing.*;
 import javax.swing.border.EmptyBorder;
+import javax.xml.bind.JAXBException;
 import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Path;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.io.*;
 
 /**
  * Main frame of the PDFA Conformance Checker
  *
  * @author Maksim Bezrukov
  */
+
 public class PDFValidationApplication extends JFrame {
 
+    class ExitWindowAdapter extends WindowAdapter {
+
+        @Override
+        public void windowClosing(WindowEvent e) {
+            PDFValidationApplication app = (PDFValidationApplication) e.getSource();
```

Comment:
get PDFValidationApplication object by calling PDFValidationApplication.this instead of e.getSource()